src/libMessage/Messenger.cpp

```diff
@@ -323,19 +301,61 @@ inline bool CheckRequiredFieldsProtoStateData(
 // Protobuf <-> Primitives conversion functions
 // ============================================================================
 
-void AccountToProtobuf(const Account& account, ProtoAccount& protoAccount) {
-  protoAccount.set_version(account.GetVersion());
+void AccountBaseToProtobuf(const AccountBase& accountbase,
+                           ProtoAccountBase& protoAccountBase) {
+  protoAccountBase.set_version(accountbase.GetVersion());
   NumberToProtobufByteArray<uint128_t, UINT128_SIZE>(
-      account.GetBalance(), *protoAccount.mutable_balance());
-  protoAccount.set_nonce(account.GetNonce());
-  protoAccount.set_storageroot(account.GetStorageRoot().data(),
-                               account.GetStorageRoot().size);
-
-  if (account.GetCode().size() > 0) {
-    protoAccount.set_codehash(account.GetCodeHash().data(),
-                              account.GetCodeHash().size);
-    protoAccount.set_code(account.GetCode().data(), account.GetCode().size());
+      accountbase.GetBalance(), *protoAccountBase.mutable_balance());
+  protoAccountBase.set_nonce(accountbase.GetNonce());
+  if (accountbase.GetCodeHash() != dev::h256()) {
+    protoAccountBase.set_codehash(accountbase.GetCodeHash().data(),
+                                  accountbase.GetCodeHash().size);
+  }
+  if (accountbase.GetStorageRoot() != dev::h256()) {
+    protoAccountBase.set_storageroot(accountbase.GetStorageRoot().data(),
+                                     accountbase.GetStorageRoot().size);
+  }
+}
+
+bool ProtobufToAccountBase(const ProtoAccountBase& protoAccountBase,
+                           AccountBase& accountBase) {
+  accountBase.SetVersion(protoAccountBase.version());
```

Comment:
If you agree to change all fields to optional, make sure to add the `CheckRequiredFieldsXXX` here, before accessing version, etc.